### PR TITLE
codegen: codegen generic event and response interfaces + usability improvements for helpers.pb.go

### PR DESCRIPTION
**Pull request by willfindlay**
Slight refactors to make the codegen API more usable, especially around helpers.pb.go. In particular, we remove all the blank interfaces and replace them with sane types that enable compile-time type checking.

Please see individual commit messages.

### CODE REVIEW

**Review by michi-covalent — APPROVED** [review]
🚢

**Comment by jrfastab** [comment]
@willfindlay needs a rebase thanks!

**Comment by willfindlay** [comment]
@jrfastab done

**Comment by willfindlay** [comment]
Bad rebase, fixing now.

**Comment by willfindlay** [comment]
@jrfastab should be good to go

File: api/v1/tetragon/codegen/eventchecker/eventchecker.pb.go [hunk 1]
```diff
@@ -269,7 +269,7 @@ func CheckerFromResponse(response *tetragon.GetEventsResponse) (EventChecker, er
 }
 
 // Event is an empty interface used for events like ProcessExec, etc.
-type Event interface{}
+type Event tetragon.Event
```
Comment (kkourt) [inline api/v1/tetragon/codegen/eventchecker/eventchecker.pb.go:272]: Nice!

**Review by kkourt — APPROVED** [review]

**Pull request merged**